components/calendar/calendar.js

```diff
@@ -0,0 +1,268 @@
+import React from 'react';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+import DaysPanel from './panels/days';
+import calendarConstants from './constants/calendar';
+
+const {
+  CALENDAR_MOVE_TO_NEXT,
+  CALENDAR_MOVE_TO_PREVIOUS,
+  CALENDAR_SELECTION_TYPE_RANGE,
+} = calendarConstants;
+
+const { PropTypes } = React;
+
+export default class Calendar extends React.Component {
+  constructor(props) {
+    super();
+
+    const { initialDates, maxDate, minDate } = props;
```

Comment:
is it possible by any chance that props will be changed during the lifetime of the component? if so, should you handle it somehow by `componentWillReceiveProps`?